### PR TITLE
Adds the raw powershell equivalents for github actions commands

### LINKED ISSUE #15355: The Action workflow commands section assumes bash, but pwsh is now the default shell on windows
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://github.com/github/docs/blob/main/content/actions/using-workflows/workflow-commands-for-github-actions.md

### What part(s) of the article would you like to see updated?

The default shell on Windows ir now pwsh, all of the examples are only for bash. And there are a few caveats. Attached PR adds PowerShell examples

Ideally the page would be split across technology [bash, pwsh, powershell, javascript] and not [linux, mac, windows] cause, like bash, pwsh is also available on all platforms. I couldn't find how to easily make a tech selector for things other than targetPlatform.

### Additional information

The default shell on Windows ir now pwsh, all of the examples are only for bash. And there are a few caveats.

Ideally the page would be split across technology [bash, pwsh, powershell, javascript] and not [linux, mac, windows] cause, like bash, pwsh is also available on all platforms. I couldn't find how to easily make a tech selector for things other than targetPlatform.

### PR DESCRIPTION
**Pull request by jessehouwing**
Closes: #15355

### Why:

The default shell on Windows ir now `pwsh`, all of the examples are only for `bash`. And there are a few caveats.

Ideally the page would be rplit across technology `[bash, pwsh, powershell, javascript]` and not `[linux, mac, windows]` cause, like `bash`, `pwsh` is also available on all platforms. I couldn't find how to easily make a tech selector for things other than `targetPlatform`.

### What's being changed:

This adds a `pwsh` example to each place where action command examples are shown in `bash`.

Staging preview:
- "[Workflow commands for GitHub Actions](https://docs-preview-15354-1e2813.eastus.azurecontainer.io/en/actions/using-workflows/workflow-commands-for-github-actions)"

### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Work in progress

 - [x] Add some kind of platform / tool selector
 - [x] Verify examples for any caveats that may be missing
 - [x] Verify the New-Guid approach is valid alternative over getting the sha256 hash of the github token.
 - [x] Does there need to be a link to the default shell per platform in here anywhere?

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@jessehouwing 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Review by fullofhope336 — COMMENTED** [review]
> ``

**Comment by jessehouwing** [comment]
Some issues with this way of displaying:

For YAML blocks setting the highlighting to `powershell` the wrong keywords are highlighted.
image

And it's unclear that this is PowerShel Core 6+ (pwsh) instead of Windows PowerShell 5 and below.

In the legacy example the difference is even more important.
image

**Comment by martin389** [comment]
@jessehouwing -- Thanks for the feedback!

> For YAML blocks setting the highlighting to powershell the wrong keywords are highlighted.

We don't currently have a way to correctly render multiple languages within the same codeblock 🤔 We're finding that the codeblock titles for "Shell" and "PowerShell" are useful for letting us distinguish them, and saves us from having to keep each example in a separate subsection -- this would occupy a lot more screen space and make it difficult for users to scan through the content. 

> And it's unclear that this is PowerShel Core 6+ (pwsh) instead of Windows PowerShell 5 and below.

Ahh true, I'll look at some options for distinguishing these 👍

**Review by av8465 — COMMENTED** [review]
Thanks

**Comment by martin389** [comment]
Status update: this is in the queue for peer review

**Review by skedwards88 — COMMENTED** [review]
🚀 Thank you for adding these examples!

@martin389 I have some concerns over using `bash` or `PowerShell` highlighting for YAML code examples. I think you have a good point that the language label makes it easier to scan. However, I think the unconventional highlighting can throw a user off and the language label can mislead a new user who doesn't readily recognize syntax without being told what language it is.

For now, I think it would be better to use YAML highlighting for YAML code examples and instead use headers or text to distinguish between bash and PowerShell for YAML examples that are bash or PowerShell specific.

Long term, I think it would be worth asking a docs PM for their opinion about using the tool selector for something like this since a user will generally only care about one language (e.g. bash vs powershell vs legacy powershell). As we add more examples (e.g. if we add legacy powershell or JavaScript examples), this solution would help keep the article from growing too long. (As a side note, it is very simple to add a new tool to the tool selector.)

**Comment by martin389** [comment]
> For now, I think it would be better to use YAML highlighting for YAML code examples and instead use headers or text to distinguish between bash and PowerShell for YAML examples that are bash or PowerShell specific.

Thanks @skedwards88 -- I've updated the PR to take this approach instead.

**Review by jessehouwing — COMMENTED** [review]
There's a few examples for PowerShell that use `echo "value" >>` where the `echo` can be dropped. Whenever a value is redirected to a file there is no need to add `write-output` or `echo` (which is shorthand for Write-Output)

**Comment by jessehouwing** [comment]
I'll make these updates in a minute.

**Comment by jessehouwing** [comment]
There were a whole lot of blocks marked YAML how, that weren't YAML. A number were plaintext, others bash and some pwsh. That threw off syntax highlighting even further.

**Comment by martin389** [comment]
Thanks for the updates @jessehouwing ✨

**Comment by martin389** [comment]
For the next step: this is ready for @skedwards88 to review our changes 🚀

**Comment by ethomson** [comment]
Thanks @jessehouwing.  I think that this makes sense overall.

>  Add some kind of platform / tool selector

Yes, this would be awesome.  I think that my concern here is that it really adds a bit of bulk to the pages.  I'm not sure we have the toggle where it would only show bash _or_ powershell.  (I know that we have something kind of like it though.)  @martin389 is this a thing that exists?

>  Verify the New-Guid approach is valid alternative over getting the sha256 hash of the github token.

Yes.  In fact, I really don't like `echo token` here...  I realize that it's never hitting the console, but it still makes my spidey sense tingle.  macOS and Ubuntu both have `uuidgen` installed, can we use that?  It both maps nicely to what you're doing in powershell _and_ doesn't touch the token.

**Review by skedwards88 — COMMENTED** [review]
Thanks this update! I think the bash vs powershell toggle makes the article more readable (and lets the user focus on the relevant content).

Will we want to distinguish between `shell: pwsh` and `shell: powershell` in the tool selector in the future, or is displaying both at the same time preferred?

Can you also add PowerShell and bash to the usage docs (https://github.com/github/docs/blob/main/contributing/content-markup-reference.md#tool-tags)?

**Review by heiskr — DISMISSED** [review]
Code changes seem fine to me

**Comment by martin389** [comment]
Thanks for the review @skedwards88 

> Will we want to distinguish between `shell: pwsh` and `shell: powershell` in the tool selector in the future, or is displaying both at the same time preferred?

I don't think we need to distinguish these with a seperate tool tag, since there are only a couple of places where we need to point out the differences in UTF-8 encoding.

> Can you also add PowerShell and bash to the usage docs (https://github.com/github/docs/blob/main/contributing/content-markup-reference.md#tool-tags)?

Done 👍

File: content/actions/using-workflows/workflow-commands-for-github-actions.md [hunk 1]
```diff
@@ -354,41 +592,103 @@ steps:
...
+```yaml{:copy}
+steps:
+  - name: Set the value in pwsh
+    id: step_one
+    run: |
+      "JSON_RESPONSE<<EOF" >> $env:GITHUB_ENV
+      (Invoke-WebRequest -Uri "https://example.lab").Content >> $env:GITHUB_ENV
+      "EOF" >> $env:GITHUB_ENV
+    shell: pwsh
+```
```
[the remark this answers is not in the document]
Comment (martin389) [reply content/actions/using-workflows/workflow-commands-for-github-actions.md:653]: Ah yes, missed this. Added a commit now 👍

**Comment by martin389** [comment]
Investigating the staging errors.

**Comment by emilyistoofunky** [comment]
From @skedwards88:
> Long term, I think it would be worth asking a docs PM for their opinion about using the tool selector for something like this since a user will generally only care about one language (e.g. bash vs powershell vs legacy powershell). As we add more examples (e.g. if we add legacy powershell or JavaScript examples), this solution would help keep the article from growing too long. (As a side note, it is very simple to add a new tool to the tool selector.)

@martin389 We have an issue that @skedwards88 recently opened to define guidelines for using the tool picker that we should address before adding more tools to it, so that we have a shared understanding for when it's appropriate for the platform overall to include a tool. Feel free to set up a meeting with me, @skedwards88, and @ethanpalm if you'd like to help move that forward.

**Comment by martin389** [comment]
Closing and reopening this PR, for CI purposes

**Comment by martin389** [comment]
Status update: docs engineering are investigating the staging issue

**Comment by jessehouwing** [comment]
For some reason this header doesn't render as it's supposed to on the preview environment:

```
## Echoing command outputs
```
image

**Comment by martin389** [comment]
> For some reason this header doesn't render as it's supposed to on the preview environment:
> 
> ```
> ## Echoing command outputs
> ```

Looks like your commit fixed it 😄 🎊

**Comment by martin389** [comment]
Following up on the content plan consideration described here: https://github.com/github/docs/pull/15354#issuecomment-1059549730

**Comment by martin389** [comment]
> @martin389 We have an issue that @skedwards88 recently opened to define guidelines for using the tool picker that we should address before adding more tools to it, so that we have a shared understanding for when it's appropriate for the platform overall to include a tool.

Thanks @emilyistoofunky, I've added the reasoning for the two new tools to @skedwards88's issue 👍

**Review by skedwards88 — DISMISSED** [review]

**Comment by dkfzhf11** [comment]
https://secure4.store.apple.com/kr/shop/signIn/account?ssi=1AAABf5nPY9wgJizik-7qc-AAl9ESfEuiQX0vbs_XJXxDZEAQQwWv-pYAAAA2aHR0cHM6Ly9zZWN1cmU0LnN0b3JlLmFwcGxlLmNvbS9rci9zaG9wL2FjY291bnQvaG9tZXx8AAIBesNzYFufi1HFI_HymBHPUHgKuz8KWqBMU2Ew3UamrhY

**Comment by martin389** [comment]
This is looking ready to ship 🚢 🎊 Just doing some final checks.

**Review by martin389 — APPROVED** [review]

**Comment by martin389** [comment]
@jessehouwing - Thanks again for this great PR, PowerShell users are definitely going to appreciate this update ✨

**Comment by martin389** [comment]
Changes are now live: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions ⚡

**Pull request merged**